### PR TITLE
Add new custom block "DetailsBlock" with expanding answers and demonstrate it in a BreadPage and in a new FAQ page.

### PR DESCRIPTION
**Pull request by gzark1**
@lb- suggested adding a `DetailsBlock` to showcase a custom block with its own HTML template. The `DetailsBlock` includes three fields: `summary` (CharField), `content` (RichTextField), and `open` (BooleanField, defaulting to open). This block is ideal for creating an FAQ page with expanding answers and allows for the use of semantic and modern HTML.

**Changes Made:**

1. Created a `DetailsBlock` class inheriting from `StructBlock` with the specified fields.
2. Added the `DetailsBlock` to the StreamField used by other pages for their content, providing them with the option to include a details block.

image

3. Created a template, `details_block.html`, for rendering the `DetailsBlock`.
4. Added CSS selectors for the `DetailsBlock` summary and content in the main CSS file.
5. Demonstrated the usage of the `DetailsBlock` in a bread page. 

image

6. Implemented a standalone StreamField with only the `DetailsBlock` option, showcased in a new `FAQ` page of the `StandardPage` type.

image

image

7. Created sample data for both the bread page and the FAQ page, stored in `bakerydemo/base/fixtures/bakerydemo.json`.

8. Added the `FAQ` page to the navigation menu.

image

### CODE REVIEW

File: bakerydemo/base/blocks.py [hunk 1]
```diff
@@ -77,3 +94,13 @@ class BaseStreamBlock(StreamBlock):
         icon="media",
         template="blocks/embed_block.html",
     )
+    details_block = DetailsBlock()
+
+
+# StreamBlocks for Details
```
Comment (lb-) [inline bakerydemo/base/blocks.py:100]: This is probably not needed, you can use a single block easily with StreamField

File: bakerydemo/base/fixtures/bakerydemo.json [hunk 2]
```diff
@@ -96,9 +96,9 @@
     "model": "base.genericsettings",
     "pk": 1,
     "fields": {
-      "twitter_url": "https://twitter.com/wagtailcms",
-      "github_url": "https://github.com/wagtail/wagtail",
-      "organisation_url": "https://wagtail.org/"
+      "twitter_url": "",
+      "github_url": "",
+      "organisation_url": ""
```
Comment (lb-) [inline bakerydemo/base/fixtures/bakerydemo.json:101]: I don't think this removal is intentional.

File: bakerydemo/base/fixtures/bakerydemo.json [hunk 3]
```diff
@@ -1300,6 +1300,118 @@
...
+    "model": "wagtailsearchpromotions.querydailyhits",
+    "pk": 6,
+    "fields": {
+      "query": 6,
+      "date": "2023-06-23",
+      "hits": 1
+    }
+  },
+  {
+    "model": "wagtailsearchpromotions.querydailyhits",
```
Comment (lb-) [inline bakerydemo/base/fixtures/bakerydemo.json:1407]: I don't think we need to add these to the fixtures

File: bakerydemo/templates/blocks/details_block.html [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<details class="details-block" {% if self.open %}open{% endif %}>
+    <summary>{{ self.summary }}</summary>
+    <div class="details-content">
+        {{ self.content }}
+    </div>
+</details>
```
Comment (lb-) [inline bakerydemo/templates/blocks/details_block.html:6]: Please add a line break at the end of this file.

**Review by lb- — CHANGES_REQUESTED** [review]
A few changes, I think the way the fixtures have been changed needs some work.

File: bakerydemo/base/blocks.py (hunk 1 above)
Comment (gzark1) [reply bakerydemo/base/blocks.py:100]: So you suggest not adding the DetailsBlock option to the "old" `BaseStreamBlock` , just to the "new" `DetailsStreamBlock` that I created?

**Comment by lb-** [comment]
@gzark1 can you close this PR if the newer one is preferred

**Comment by thibaudcolas** [comment]
Closing in favour of #425.

**Pull request closed without merging**